clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/LegacyFailureDetectionProtocolResourceDefinitionRegistrar.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.clustering.jgroups.subsystem;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.RequirementServiceBuilder;
+import org.jboss.as.network.ClientMapping;
+import org.jboss.as.network.SocketBinding;
+import org.jboss.dmr.ModelNode;
+import org.jgroups.protocols.FD_SOCK;
+import org.wildfly.clustering.jgroups.spi.ProtocolConfiguration;
+import org.wildfly.clustering.jgroups.spi.ChannelFactoryConfiguration;
+import org.wildfly.subsystem.resource.operation.ResourceOperationRuntimeHandler;
+import org.wildfly.subsystem.service.ServiceDependency;
+
+/**
```

Comment:
Some explanation of the purpose of this would help readability.

Reply:
Fixed.